TT-13174 Operator and Sync and Charts release (#5460)
TykTechnologies/tyk-docs: tyk-docs/content/product-stack/tyk-charts/release-notes/version-2.1.md

---
title: Tyk Charts 2.1 Release Notes
description: "Release notes documenting updates, enhancements and changes for Tyk Charts versions within the 2.1 series."
tags: ["Tyk Charts", "Release notes", "changelog", "v2.1" ]
---

****Open Source** ([Mozilla Public License](https://github.com/TykTechnologies/tyk/blob/master/LICENSE.md))**

**This page contains all release notes for version 2.1.X displayed in a reverse chronological order**

### Support Lifetime
<!-- Required. replace X.Y with this release and set the correct quarter of the year -->
Our minor releases are supported until our next minor comes out. 

---

> ## 2.1.0 Release Notes

Author: This is based on the template at /assets/others/release-notes-template.md :(

There are no deprecation in this release.

<!-- Optional section!
Used to share and notify users about our plan to deprecate features, configs etc. 
Once you put an item in this section, we must keep this item listed in all the following releases till the deprecation happens
##### Future deprecations. -->

#### Upgrade instructions
<!-- Required. For patches release (Z>0) use this: -->
For users currently on v2.0.x, we strongly recommend promptly upgrading to the latest release. 
<br/>
<!-- Go to the [Upgrading Tyk](#upgrading-tyk) section for detailed upgrade Instructions.
-->
You can use helm upgrade to upgrade your release

```bash
helm repo add tyk-helm https://helm.tyk.io/public/helm/charts/
helm repo update

helm upgrade [RELEASE_NAME] tyk-helm/[CHART_NAME]
```

#### Release Highlights
<!-- Required. Use similar ToV to previous release notes. For example for a patch release: -->

##### Updated default Tyk versions
Tyk Charts 2.1 will install the following Tyk component versions by default.
- Tyk Gateway v5.3.6
- Tyk Dashboard v5.3.6
- Tyk Pump v1.11.0
- Tyk MDCB v2.7.1
- Tyk Developer Portal v1.10.0

> - Tyk Operator v1.0.0

Author: updated

Here it is important to explain the benefit of each changelog item. As mentioned by James in a previous Slack message (https://tyktech.slack.com/archives/C044R3ZTN6L/p1686812207060839?thread_ts=1686762128.651249&cid=C044R3ZTN6L):
"...it is important to document the customer impact for the work delivered, so we can share it with prospects/install base. For example:
"New Chart delivers x and y benefit to a and b customer use cases. The business impact for them will be this and that" -->

##### Added
<!-- This section should be a bullet point list of new features. Explain:

- The purpose of the new feature
- How does the new feature benefit users?
- Link to documentation of the new feature
- For OSS - Link to the corresponding issue if possible on GitHub to allow the users to see further info.

Each change log item should be expandable. The first line summarises the changelog entry. It should be then possible to expand this to reveal further details about the changelog item. This is achieved using HTML as shown in the example below. -->

<ul>

<li>
<details>

> <summary>Gateway and Dashboard: new LoadBalancer IP service parameters</summary>

Reviewer: ```suggestion
<summary>Ability to specify static IP for Kubernetes LoadBalancer service</summary>
```

Added optional `loadBalancerIP` parameter that allows users to set a static IP for Tyk Gateway and Dashboard services when using the `LoadBalancer` service type. This update provides enhanced control over IP configuration, useful for network stability in environments with multiple load balancers.

Tyk gateway service configuration:
- `tyk-gateway.gateway.service.loadBalancerIP` (default to "")

Tyk Dashboard service configuration:
- `tyk-dashboard.dashboard.service.loadBalancerIP` (default to "")

</details>
</li>

<li>
<details>

> <summary>Dashboard: container port configuration</summary>

Reviewer: ```suggestion
<summary>Ability to configure Dashboard container port</summary>
```

Enables specifying an alternate port for the container while using standard ports in the service. This option resolves permission issues associated with restricted ports, such as port 443, within containers.

</details>
</li>

<li>
<details>

> <summary>Operator: license key configuration</summary>

Reviewer: ```suggestion
<summary>From v1.0 Tyk Operator Requires License Key</summary>
```
